### PR TITLE
Feature al 200 desing and implement functions service

### PR DESCRIPTION
**Pull request by atanas-arnaudov-prime**

### CODE REVIEW

File: src/Lykke.AlgoStore.CSharp.Algo.Core/Functions/AbstractFunction.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    {
+        public class FunctionInvocationException : Exception
+        {
+            public FunctionParamsBase FunctionParameters { get; set; }
+
+            public FunctionInvocationException(string message, Exception innerException)
+                : base(message, innerException) { }
+        }
+
+        public class AddNewVlaueException : FunctionInvocationException
```
Comment (mjoksa) [inline src/Lykke.AlgoStore.CSharp.Algo.Core/Functions/AbstractFunction.cs]: Change AddNewVlaueException to AddNewValueException

File: src/Lykke.AlgoStore.CSharp.Algo.Core/Functions/AbstractFunction.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        {
+            public Candle ValueInvokedWith { get; set; }
+
+            public AddNewVlaueException(string message, Exception innerException)
```
Comment (mjoksa) [inline src/Lykke.AlgoStore.CSharp.Algo.Core/Functions/AbstractFunction.cs]: Same here :) - Change AddNewVlaueException to AddNewValueException

**Review by mjoksa — CHANGES_REQUESTED** [review]
Please address small changes in naming :) and then we can merge this :)

File: src/Lykke.AlgoStore.CSharp.Algo.Core/Functions/AbstractFunction.cs (hunk 2 above)
Comment (atanas-arnaudov-prime) [reply src/Lykke.AlgoStore.CSharp.Algo.Core/Functions/AbstractFunction.cs]: Thanks, applied!

**Pull request merged**